subprojects/core/src/main/java/org/gradle/execution/taskgraph/DefaultTaskExecutionGraph.java

```diff
@@ -216,7 +217,7 @@ public boolean hasTask(String path) {
 
     public List<Task> getAllTasks() {
         ensurePopulated();
-        return taskExecutionPlan.getTasks();
+        return ImmutableList.copyOf(taskExecutionPlan.getTasks());
```

Comment:
I'd cache this once it's been used once.

Reply:
Caching now.